tests/test_querysets/test_reservation_querysets.py

```diff
@@ -0,0 +1,316 @@
+from datetime import UTC, datetime
+
+import pytest
+from django.utils.timezone import get_current_timezone
+
+from opening_hours.utils.reservable_time_span_client import TimeSpanElement
+from reservation_units.models import ReservationUnit
+from reservations.models import Reservation
+from tests.factories import (
+    ReservationFactory,
+    ReservationUnitFactory,
+    ResourceFactory,
+    SpaceFactory,
+    UnitFactory,
+)
+
+# Applied to all tests
+pytestmark = [
+    pytest.mark.django_db,
+    pytest.mark.usefixtures("_disable_elasticsearch"),
+]
+
+DEFAULT_TIMEZONE = get_current_timezone()
+
+
+def _datetime(year=2023, month=5, day=1, hour=0, minute=0) -> datetime:
+    # Convert to UTC to match timezone returned by GQL endpoint
+    return datetime(year, month, day, hour, minute, tzinfo=DEFAULT_TIMEZONE).astimezone(UTC)
+
+
+def _create_test_reservations_for_all_reservation_units() -> None:
+    """
+    Create test reservations for all ReservationUnits.
+
+    Use the ReservationUnit's creation order as the minute for the Reservation's begin datetime.
+    Due to the TimeSpanElements returned by `as_closed_time_spans` not including any identifying information about
+    the reservations, we need to do this is so that the TimeSpanElements can be distinguished from each other,
+    which is a requirement for testing and helps a lot with debugging.
+    """
+    for i, reservation_unit in enumerate(ReservationUnit.objects.order_by("pk").all(), start=1):
+        ReservationFactory.create(
+            begin=_datetime(minute=i),
+            end=_datetime(minute=0),
+            reservation_unit=[reservation_unit],
+            user=None,
+        )
+
+
+def _validate_time_spans(
+    all_closed_time_spans: dict[int, list[TimeSpanElement]],
+    reservation_unit_id: int,
+    related_reservation_unit_ids: list[int],
+) -> None:
+    """
+    Validate that the given ReservationUnit has the expected closed time spans generated from related ReservationUnits.
+
+    :param all_closed_time_spans: The result of `Reservation.objects.as_closed_time_spans`.
+    :param reservation_unit_id: The id of the ReservationUnit to validate.
+    :param related_reservation_unit_ids: The ids of the ReservationUnits that should be related to the ReservationUnit.
+    """
+    closed_time_spans: set[TimeSpanElement] = {
+        TimeSpanElement(
+            start_datetime=reservation.buffered_begin,
+            end_datetime=reservation.buffered_end,
+            is_reservable=False,
+        )
+        for reservation in Reservation.objects.with_buffered_begin_and_end().filter(
+            reservation_unit__pk__in=related_reservation_unit_ids + [reservation_unit_id]
+        )
+    }
+
+    assert all_closed_time_spans.get(reservation_unit_id) == closed_time_spans
+
+
+def test__get_affecting_reservations__only_resources():
+    # Resource 1 <- ReservationUnit 1, 2
+    # Resource 2 <- ReservationUnit 3, 4
+    # Resource 2 <- ReservationUnit 5
```

Comment:
Typo, should read: # Resource **3** <- ReservationUnit 5

Reply:
fixed